Wrong return types

pabra opened the pull request:
`not` and `toBool` had wrong return types

Commit:
add type tests for formulas

Commit:
fix return types

denistakeda commented on src/impl/Formulas.ts:
```diff
@@ -37,10 +37,10 @@ export function byIndex<T>(
 }
 
-export function toBool(source: Cell<unknown>): Value<boolean> {
+export function toBool(source: Cell<unknown>): FormulaCell<boolean> {
```
We probably should use the `formula` function instead of creation of a `FormulaCell` object. This way we can use these functions with `Value` and return `Value` as well

pabra replied:
Those "prepared formulas" (`not`, `toBool`, etc...) now use all `formula` function instead of `new FormulaCall`.  I also added more type tests to ensure returned types for various input types.

How about adding a GitHub action to automatically run tests on each commit/pull request?

denistakeda replied:
Yes, this sounds as a good idea

Commit:
make helper formulas generic and accept plain values as well as Cells

Commit:
add more type tests - specially for record types passed to `field`

The reason to pick about Record or object is, for an object with known
(literal) keys, `field` should return the specific key of that entry. In
case of an Record, `field` should always return the value type OR (`|`)
undefined because keys are not known.

Commit:
combine test commands
